keras/backend/theano_backend.py

```diff
@@ -375,8 +377,8 @@ def gradients(loss, variables):
 
 # CONTROL FLOW
 
-def rnn(step_function, inputs, initial_states,
-        go_backwards=False, masking=True):
+def rnn(step_function, inputs, output_dim, initial_states,
```

Comment:
The `output_dim` / `initial_output` business should definitely be eliminated by all means necessary.